internal/worker.go

```diff
@@ -249,43 +179,18 @@ func IsReplayDomain(dn string) bool {
 }
 
 // NewWorker creates an instance of worker for managing workflow and activity executions.
-// domain - the name of the temporal domain.
-// taskList 	- is the task list name you use to identify your client worker, also
-// 		  identifies group of workflow and activity implementations that are hosted by a single worker process.
-// options 	-  configure any worker specific options like hostPort, logger, metrics, identity.
+// client   - client created with client.NewClient().
+// taskList - is the task list name you use to identify your client worker, also
+//            identifies group of workflow and activity implementations that are hosted by a single worker process.
+// options 	- configure any worker specific options.
 func NewWorker(
+	client Client,
 	taskList string,
 	options WorkerOptions,
-) (*AggregatedWorker, error) {
-
-	if options.MetricsScope == nil {
-		options.MetricsScope = tally.NoopScope
-		options.Logger.Info("No metrics scope configured for temporal worker. Use NoopScope as default.")
+) *AggregatedWorker {
+	workflowClient, ok := client.(*WorkflowClient)
```

Comment:
This is another shortcut I plan to refactor later. I didn't touch `Client` interface.

Actually probably it is better for `NewClient` to return `WorkflowClient` struct not `Client` interface. And `NewWorker` should also depend on `WorkflowClient` not `Client`.

Reply:
This is not ideal.  Let's open an issue to clean this up later.